lib/racecar/producer.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require "racecar/message_delivery_error"
+
+
+module Racecar
+  class Producer
+
+    def initialize(config: nil, logger: nil, instrumenter: NullInstrumenter)
+      @config = config
+      @logger = logger
+      @delivery_handles = []
+      @instrumenter = instrumenter
+      @batching = false
+    end
+
+
+    # fire and forget - you won't get any guarantees or feedback from 
+    # Racecar on the status of the message and it won't halt execution
+    # of the rest of your code.
+    def produce_async(value:, topic:, **options)
+      handle = internal_producer.produce(payload: value, topic: topic, **options)
+      @delivery_handles << handle if @batching
+
+      nil
+    end
+
+    # synchronous message production - will wait until the delivery handle succeeds, fails or times out.
+    def produce_sync(value:, topic:, **options)
+      begin
+        handle = internal_producer.produce(payload: value, topic: topic, **options).wait(max_wait_timeout: 60, wait_timeout: 0.1)
+      rescue Rdkafka::AbstractHandle::WaitTimeoutError => e
+        partition = MessageDeliveryError.partition_from_delivery_handle(handle)
+        @logger.debug "Still trying to deliver message to (partition #{partition})... (will try up to Racecar.config.message_timeout)"
```

Comment:
```suggestion
        @logger.warn "Still trying to deliver message to (partition #{partition})... (will try up to Racecar.config.message_timeout)"
```

I think this is significant enough, and should happen rarely enough, that it warrants `WARN`.